### ISSUE TITLE
but(FormFiled): Floating label hide/show not responsive when set to outline appearance.

### ISSUE THREAD

**Issue by Cole-Jaeger** [angular/components#29939]
### Is this a regression?

- [ ] Yes, this behavior used to work in the previous version

### Description

When you have a mat form field and the mat-label is conditional. If the view starts with the label and gets remove with @if in template it is perfectly fine. Will disappear and reappear. If the view starts with-out the mat label rendered, the label will get rendered but the component, 'MatFormFieldNotchedOutline" still has the class, 'mdc-notched-outline--no-label', so the display is set to none.

Looks as though if the mat-label is not there on view init it is not responsive.

The behavior works fine with fill appearance.

### Reproduction

StackBlitz link: 
Steps to reproduce:
1. 
2.

### Expected Behavior

When mat-label gets rendered in even after init, i.e. Using a structural directive or decorator such *ngIf or @if. The label should show no matter the appearance. 

### Actual Behavior

When you have the appearance set to outline and the view is initially rendered without the label it is not responsive and will not show up.

### Environment

- Angular: 18.2.10
- CDK/Material: 18.2.10
- Browser(s): Edge, Chrom
- Operating System (e.g. Windows, macOS, Ubuntu): Windows

**Comment by Totati** [comment]
Duplicate of #26579

**Comment by Cole-Jaeger** [comment]
I don't think this is a duplicate of that other bug. That one is only happening on fill appearance while this is happening on the outline appearance. This is a specific issue of what it seems like is in the "ngAfterViewInit" in the MatFormFieldNotchedOutline component. A component that does not even get added when appearance is set to fill.

**Comment by Cole-Jaeger** [comment]
Absolutely @amysorto ! 
<https://stackblitz.com/edit/components-issue-starter-gjragy?file=src%2Fmain.ts>

This is a super simple example in my app the "showLabel" is based off of a media break point but this example just uses a simple toggle.

Please let me know if you need anything else

Thank you.

**Comment by amysorto** [comment]
@Cole-Jaeger thanks for the link, I am able to reproduce the issue

**Comment by vladsadretdinov** [comment]
Bug still exists in 19 version too.

```
"@angular/cdk": "19.0.4",
"@angular/material": "19.0.4",
```

As temporary solution we use this approach (@Cole-Jaeger maybe you are interested in this)

```js
<mat-form-field 
    appearance=outline 
    [ngClass]="{'mdc-notched-outline--no-label': !showLabel}"
>
    <mat-label>
        value
    </mat-label>

    <input matInput/>
</mat-form-field>
```

**Comment by mistrykaran91** [comment]
Hi! Is anybody working on it? Otherwise, I could try to fix it if you don't mind.
